app/services/promo_registration_getter.rb

```diff
@@ -18,13 +18,13 @@ def perform
     response = connection.get do |request|
       request.headers["Authorization"] = api_authorization_header
       request.headers["Content-Type"] = "application/json"
-      request.url("/api/1/promo/publishers?channel=#{@channel.channel_id}")
+      request.url("/api/1/promo/publishers?channel=#{@channel.details.channel_identifier}")
     end
     registrations =  JSON.parse(response.body)
     referral_code = referral_code_for_promo_id(registrations)
     
     if should_update_channel_owner_on_promo_server(registrations)
-      PromoChannelOwnerUpdater.new(publisher_id: @publisher.id, referral_code: referral_code).perform
+      PromoChannelOwnerUpdater.new(publisher_id: @publisher.owner_id, referral_code: referral_code).perform
```

Comment:
Should this be `@publisher.owner_identifier`? Or maybe we should pass `@publisher.id` and then find the corresponding `@publisher` in `PromoChannelOwnerUpdater`.

Looking at `PromoChannelOwnerUpdater`, I see we're setting `@publisher_id` to this value. However, in some places `@publisher.id` is also referenced, without instantiating `@publisher`.

Reply:
@dgeb Good catch, you are correct on both counts.